toolchain/lower/file_context.h

```diff
@@ -45,6 +49,13 @@ class FileContext {
                           llvm::Module& llvm_module,
                           llvm::DIBuilder& di_builder) -> llvm::DICompileUnit*;
 
+  // Creates the Clang `CodeGenerator` to generate LLVM module from imported C++
+  // code. Returns null when not importing C++. Can be called only after the
+  // following data members are initialized: `llvm_context_`, `llvm_module_`,
+  // `fs_`, `cpp_ast_`, `cpp_header_search_options_`,
+  // `cpp_preprocessor_options_`, `cpp_code_gen_options_`, .
```

Comment:
The last part of this comment seems a bit redundant, given that most of those members are initialized by the constructor. Also if there's a precondition here, it'd be better for it to be user-facing -- that is, saying which member functions of this class must be called first, not which data members must be initialized.

Reply:
The reason I did it this way was because `CreateCppCodeGenerator()` is called within the constructor, when initializing `cpp_code_generator_`.
I could create it in `Init()` instead, before it is initialized.
WDYT?

Reply:
I see. Then maybe you could add to the comment that this function is called from the constructor while still initializing some of the class members? That would have helped me understand what's going on.

Alternatively, could you call this from the constructor body rather than from the member initializers, so it's only called once the invariants of the class are already set up?

Reply:
Thanks!
Calling the constructor body.